velox/expression/tests/ExprTest.cpp

```diff
@@ -2109,24 +2116,57 @@ TEST_P(ParameterizedExprTest, memo) {
   auto rowType = ROW({"c0"}, {base->type()});
   auto exprSet = compileExpression("c0[1] = 1", rowType);
 
-  auto result = evaluate(
+  auto [result, stats] = evaluateWithStats(
       exprSet.get(), makeRowVector({wrapInDictionary(evenIndices, 100, base)}));
   auto expectedResult = makeFlatVector<bool>(
       100, [](auto row) { return (8 + row * 2) % 3 == 1; });
   assertEqualVectors(expectedResult, result);
+  VELOX_CHECK_EQ(stats["eq"].numProcessedRows, 100);
+  VELOX_CHECK(base.unique());
 
-  result = evaluate(
+  // After this results would be cached
+  std::tie(result, stats) = evaluateWithStats(
```

Comment:
nit: auto [result, stats] = evaluateWithStats(...) to be consistent with the above?

Reply:
`auto [result, stats]` is used to define the variables, once they have been defined `std::tie` is used to assign new values to them. Using auto will redefine them which is not allowed